control_plane/attachment_service/src/service.rs

```diff
@@ -759,6 +759,15 @@ impl Service {
             };
 
             shard_state.generation = std::cmp::max(shard_state.generation, new_gen);
+            if let Some(observed) = shard_state
+                .observed
+                .locations
+                .get_mut(&reattach_req.node_id)
+            {
+                if let Some(conf) = observed.conf.as_mut() {
+                    conf.generation = new_gen.into();
```

Comment:
Just checking: setting the observed generation to `new_gen` instead of `std::cmp::max(shard_state.generation, new_gen)` is intentional, right? If `new_gen < shard_state.generation`, then the intent state differs from the observed and should result in a reconciliation down the line.

Reply:
Right, it's intentional.

The update to shard_state.generation is done with a max() out of paranoia, as we wouldn't want it to incorrectly go backwards if somehow we were bumping generation from a Reconciler at the same time as servicing a re-attach request.  This shouldn't be possible but currently is, as alluded to in the comment directly below "/ TODO: cancel/restart any running reconciliation for this tenant..."

The update to observed reflects what we believe this pageserver has seen rather than what we believe the generation should be for the tenant -- in the event that something else had managed to increment TenantState::generation while we were running, it would not be correct to presume that the pageserver had seen that generation